cf_xarray/tests/test_accessor.py

```diff
@@ -1584,6 +1584,14 @@ def test_custom_criteria() -> None:
         ds["temperature"] = ("dim", np.arange(10))
         assert_identical(ds.cf["temp"], ds["temperature"])
 
+    # test that having a global regex expression flag later in the expression will work if
+    # regex is found
+    vocab = {"temp": {"name": "tem|(?i)temp"}}
+    ds = xr.Dataset()
+    ds["Tempblah"] = [0, 1, 2]
+    with cf_xarray.set_options(custom_criteria=vocab):
+        assert_identical(ds.cf["temp"], ds["Tempblah"])
```

Comment:
Add  one of these for regex:
https://github.com/xarray-contrib/cf-xarray/blob/0f92bc282176df0427717663a05e8869d7822eff/cf_xarray/tests/__init__.py#L69

Then

```suggestion
	if has_regex:
       	 assert_identical(ds.cf["temp"], ds["Tempblah"])
       else:
       	with pytest.raises(...):
       		ds.cf["temp"]
```